mozilla-release/mobile/android/base/java/org/mozilla/gecko/util/AppBackgroundManager.java

```diff
@@ -176,5 +180,63 @@ public void draw(Canvas canvas) {
             }
         }
     }
+
+    private static class ScreenSizedDrawable extends StateListDrawable {
+
+        private final Drawable mDrawable;
+
+        private final int mScreenHeight;
+        private final int mScreenWidth;
+
+        ScreenSizedDrawable(Drawable drawable, Context context) {
+            super();
+            this.mDrawable = drawable;
+            final DisplayMetrics displayMetrics = new DisplayMetrics();
+            ((Activity) context).getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
```

Comment:
Why don't you just store the display metrics in the constructor of the outer class and pass them here instead of this?